lib/graphql/upgrader/member.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Upgrader
+    class Member
+      def initialize(member)
+        @member = member
+      end
+
+      def upgrade
+        transformable = member.dup
+        transformable = transform_to_class transformable
+        transformable = transform_or_remove_name transformable
+        transformable = simplify_field_definition_for_easier_processing transformable
+        transformable = move_the_type_from_the_block_to_the_field transformable
+        transformable = rename_property_to_method transformable
+
+        transformable.scan(/(?:field|connection|argument) .*$/).each do |field|
```

Comment:
I think it would improve legibility if you move this entire scan portion into a method called something along the lines of `transform_fields_and_connections_and_arguments`. Then it's on the "same level" as the methods above it.

Just a minor comment though; it's your code, I trust you.

Reply:
I like the suggestion, let me try